pilot/cmd/pilot-discovery/debug_test.go

```diff
@@ -0,0 +1,163 @@
+	}
+	p.Unlock()
+}
+
+func Test_debug_run(t *testing.T) {
+	tests := []struct {
+		name              string
+		args              []string
+		pilotNotReachable bool
+		pilotStates       []pilotStubState
+		wantError         bool
+	}{
+		{
+			name: "debug with all configType does not error",
+			args: []string{"proxyID", "all"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 200, Response: "fine"},
+			},
+		},
+		{
+			name:              "debug with all configType errors when unable to reach to pilot",
+			args:              []string{"proxyID", "all"},
+			pilotNotReachable: true,
+			wantError:         true,
+		},
+		{
+			name: "debug with all configType errors when one request to pilot returns a non 200 response",
+			args: []string{"proxyID", "all"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 200, Response: "fine"},
+				{StatusCode: 404, Response: "not fine"},
+			},
+			wantError: true,
+		},
+		{
+			name: "debug with clusters configType does not error",
+			args: []string{"proxyID", "clusters"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+			},
+		},
+		{
+			name:              "debug with clusters configType errors if pilot unreachable",
+			args:              []string{"proxyID", "clusters"},
+			pilotNotReachable: true,
+			wantError:         true,
+		},
+		{
+			name: "debug with listeners configType does not error",
+			args: []string{"proxyID", "listeners"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+			},
+		},
+		{
+			name:              "debug with listeners configType errors if pilot unreachable",
+			args:              []string{"proxyID", "listeners"},
+			pilotNotReachable: true,
+			wantError:         true,
+		},
+		{
+			name: "debug with routes configType does not error",
+			args: []string{"proxyID", "routes"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+			},
+		},
+		{
+			name:              "debug with routes configType errors if pilot unreachable",
+			args:              []string{"proxyID", "routes"},
+			pilotNotReachable: true,
+			wantError:         true,
+		},
+		{
+			name: "debug with endpoints configType does not error",
+			args: []string{"proxyID", "endpoints"},
+			pilotStates: []pilotStubState{
+				{StatusCode: 200, Response: "fine"},
+			},
+		},
+		{
+			name:              "debug with endpoints configType errors if pilot unreachable",
+			args:              []string{"proxyID", "endpoints"},
+			pilotNotReachable: true,
+			wantError:         true,
+		},
+		{
+			name:      "debug with invalid configType returns an error",
+			args:      []string{"proxyID", "not-a-config-type"},
+			wantError: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
nit: should probably use shorter test names. Also a separator will be substituted for " " ... might as well make it explicit (e.g. "all_noerror")

Reply:
We don't do this anywhere else, but I agree with the name length.

Reply:
All of the pilot e2e tests do this, for example. It just makes the test output a bit more readable (at least in an IDE).